pulsar-client/src/test/java/org/apache/pulsar/client/impl/ConsumerBuilderImplTest.java

```diff
@@ -370,4 +379,102 @@ public void testTopicConsumerBuilder() {
         assertThat(topicConsumerConfigurationData.getTopicNameMatcher().matches("foo")).isTrue();
         assertThat(topicConsumerConfigurationData.getPriorityLevel()).isEqualTo(1);
     }
+
+    @Test
+    public void test() {
+        Map<String, Object> config = new HashMap<>();
+        config.put("keySharedPolicy", KeySharedPolicy.autoSplitHashRange());
+        consumerBuilderImpl.loadConf(config);
+    }
+
+    @Test
+    public void testLoadConf() {
+        ConsumerBuilderImpl consumerBuilder = new ConsumerBuilderImpl(null, Schema.BYTES);
+        consumerBuilder
+            .messageListener((consumer, message) -> {})
+            .consumerEventListener(mock(ConsumerEventListener.class))
+            .negativeAckRedeliveryBackoff(MultiplierRedeliveryBackoff.builder().build())
+            .ackTimeoutRedeliveryBackoff(MultiplierRedeliveryBackoff.builder().build())
+            .cryptoKeyReader(DefaultCryptoKeyReader.builder().build())
+            .messageCrypto(new MessageCryptoBc("ctx1", true))
+            .batchReceivePolicy(BatchReceivePolicy.builder().maxNumBytes(1).build())
+            //.keySharedPolicy(mock(KeySharedPolicy.class))
+            .messagePayloadProcessor(mock(MessagePayloadProcessor.class));
+
+        Map<String, Object> conf = new HashMap<>();
+        MessageListener messageListener = (consumer, message) -> {};
+        conf.put("messageListener", messageListener);
+        ConsumerEventListener consumerEventListener = mock(ConsumerEventListener.class);
+        conf.put("consumerEventListener", consumerEventListener);
+        RedeliveryBackoff negativeAckRedeliveryBackoff = MultiplierRedeliveryBackoff.builder().build();
+        conf.put("negativeAckRedeliveryBackoff", negativeAckRedeliveryBackoff);
+        RedeliveryBackoff ackTimeoutRedeliveryBackoff = MultiplierRedeliveryBackoff.builder().build();;
+        conf.put("ackTimeoutRedeliveryBackoff", ackTimeoutRedeliveryBackoff);
+        CryptoKeyReader cryptoKeyReader = DefaultCryptoKeyReader.builder().build();
+        conf.put("cryptoKeyReader", cryptoKeyReader);
+        MessageCrypto messageCrypto = new MessageCryptoBc("ctx2", true);
+        conf.put("messageCrypto", messageCrypto);
+        BatchReceivePolicy batchReceivePolicy = BatchReceivePolicy.builder().maxNumBytes(2).build();
+        conf.put("batchReceivePolicy", batchReceivePolicy);
+        //KeySharedPolicy keySharedPolicy = mock(KeySharedPolicy.class);
```

Comment:
Can't add KeySharedPolicy atm because it results in an exception. See https://github.com/apache/pulsar/pull/18345

Reply:
Rebased after fixing #18345